cmd/thanos/query_frontend.go

```diff
@@ -70,6 +70,9 @@ func registerQueryFrontend(app *extkingpin.App) {
 
 	cfg.CachePathOrContent = *extflag.RegisterPathOrContent(cmd, "query-range.response-cache-config", "YAML file that contains response cache configuration.", false)
 
+	cmd.Flag("query-range.compression", "Use compression in results cache. Supported values are: 'snappy' and '' (disable compression).").
```

Comment:
Can we elaborate when it is used? On response? or when put in Memcached ?

Reply:
Also, maybe `query-range.compression-type`?

Reply:
Yes, it is used before storing the response to Memcached. It doesn't make sense to use it when only fifo-cache is enabled. I should document that.

I am okay with either `query-range.compression-type` or `query-range.compression`. The flag is named to `compression` in Cortex

Reply:
> Yes, it is used before storing the response to Memcached. It doesn't make sense to use it when only fifo-cache is enabled. I should document that.

But it will work for other non FIFO backends as well, right? :thinking:

Reply:
Yes, it compresses the response and then sends it to the results cache backend. If it is fifo-cache, because snappy cache and fifo cache are in memory,  so not needed.

Reply:
Technically it could be needed -> It might reduce memory consumed as well (: 
But yea, the question is what is available right now (:

Reply:
Yeah you are right. Sorry, my mistake. It is a tradeoff between CPU and Memory & network traffic

Reply:
No need to be sorry - I think it's extremely useful. The only question is, do we need this for query range or some global cache option? (for future labels and instant)?

I think we need it for all so maybe some generic: 
```suggestion
	cmd.Flag("cache-compression", "Use compression in results cache. Supported values are: 'snappy' and '' (disable compression).").
```

Reply:
I think the option can be global. But considering other caching configs, I am not sure. This is another problem and maybe we can have a separate issue to talk about it. The main question is, can we use different cache backends and different cache configs for each endpoint we want to cache? @bwplotka @kakkoyun 

For example, it is not very good to reuse the same cache config because the response sizes are different for range queries and instant queries. They might need different limits and different max_items configs.

Reply:
Totally agree. To me, something will be the same, some things different (: